Fix a false positive for Layout/HashAlignment with table style and a multiline key (#15452)
rubocop/rubocop: changelog/fix_a_false_positive_for_layout_hash_alignment.md

> * [#15452](https://github.com/rubocop/rubocop/pull/15452): Fix a false positive (and an unsafe autocorrect that could delete part of the key) for `Layout/HashAlignment` with `EnforcedHashRocketStyle: table`/`EnforcedColonStyle: table` when a hash pair's key spans multiple lines. ([@dduugg][])

Reviewer: On master the inflated width only ever inserts whitespace, so deleting part of the key cannot actually occur. Could you keep the entry simple?

```suggestion
* [#15452](https://github.com/rubocop/rubocop/pull/15452): Fix a
  false positive for `Layout/HashAlignment` when using `EnforcedHashRocketStyle: table` and a hash key spans multiple lines. ([@dduugg][])
```

Author: Good catch, thanks. You're right that the master bug only ever inserts whitespace. Applied your suggested wording verbatim to the changelog entry, and fixed the same overstatement in the PR description (it previously implied the key-deletion risk existed on master, but that was actually something I introduced and caught myself while iterating on this fix, and never released).
